tests/drivers/uart/uart_async_api/testcase.yaml

```diff
@@ -75,3 +75,36 @@ tests:
       - CONFIG_UART_SAM0_ASYNC=y
       - CONFIG_DMA=y
     build_only: true
+  drivers.uart.async_api.nocache_mem:
+    filter: CONFIG_SERIAL_SUPPORT_ASYNC and CONFIG_CPU_HAS_DCACHE
+    harness: ztest
+    harness_config:
+      fixture: gpio_loopback
+    depends_on: gpio
+    platform_allow:
+      - nucleo_f746zg
+      - nucleo_h723zg
+    extra_configs:
+      - CONFIG_DCACHE=y
+      - CONFIG_NOCACHE_MEMORY=y
+      - CONFIG_USERSPACE=n
+  drivers.uart.async_api.nocache_mem_dt.nucleo_f746zg:
+    harness: ztest
+    harness_config:
+      fixture: gpio_loopback
+    depends_on: gpio
+    platform_allow:
+      - nucleo_f746zg
+    extra_args:
+      - DTC_OVERLAY_FILE="boards/nucleo_f746zg.overlay;boards/nucleo_f746zg_nocachemem.overlay"
```

Comment:
Do you confirm that `boards/nucleo_f746zg.overlay` is not used if not specified ?

Reply:
yes